homeassistant/components/tradfri/__init__.py

```diff
@@ -103,7 +106,8 @@ async def async_setup(hass: HomeAssistant, config: ConfigType):
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Create a gateway."""
     # host, identity, key, allow_tradfri_groups
-    tradfri_data = hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {}
+    tradfri_data: dict[str, list[Any] | None] = {}
```

Comment:
There are other instances stored in `tradfri_data` than lists, eg `APIFactory` and `APIFactory.request`.

Reply:
Just found out the hard way (VScode debugger broke down). Corrected, but I cannot run mypy locally right now so I need to see the verdict from CI.